### ISSUE TITLE
New rule `del-attr-with-constant`

### ISSUE THREAD

**Issue by janosh** [astral-sh/ruff#18338]
### Summary

new sibling for [B009](https://docs.astral.sh/ruff/rules/get-attr-with-constant) (`get-attr-with-constant`):

```py
# bad
delattr(cls, "_repr_mimebundle_")

# good
del cls._repr_mimebundle_
```

**Comment by fennr** [comment]
@MichaReiser can you assign issue to me?

**Comment by MichaReiser** [comment]
It could make sense to open a request for this rule upstream before adding it to Ruff, to avoid re-coding in the future (We can't add it to the bugbear category)
